react/components/DatePicker/index.js

```diff
@@ -19,6 +19,43 @@ class DatePicker extends Component {
     this.handleLocaleChange(props.locale)
   }
 
+  get popperModifiers() {
+    const { align, direction, size, useTime } = this.props
+
+    const isRightAligned = align === 'right'
+    const isUpwardsDirected = direction === 'up'
+
+    if (!isRightAligned && !isUpwardsDirected) {
+      return undefined
+    }
+
+    const offsetX = isRightAligned
+      ? (size === 'large' ? -91 : -136) - (useTime ? 130 : 0)
```

Comment:
where do these number come from?

Reply:
Well, the `size` part comes from pure observation, while the `useTime` one comes from the time column width, that is 130px wide.